module-3/lab-supervised-learning-feature-extraction/your-code/main.ipynb

```diff
@@ -12,14 +12,15 @@
   },
```

Comment:
- Good approach 👍 : 
```Python

def convert_string_to_numeric(s):   
        #for s in google_play['Reviews']:
        if s.endswith('M'):
            s2=int(float(s.replace('M',''))*1000000)
            return s2
        else:
            s2=int(float(s)*1000000)  
        return s2 
test_string = '4.0M'
convert_string_to_numeric(test_string) == 4000000
# Other way 
def convert_string_to_numeric(s):
    if s[-1] == 'M':
        return float(s[:-1]) * 1000000
    else:
        return float(s)
    return np.NaN
test_string = '4.0M'
convert_string_to_numeric(test_string) == 4000000
```
- In this one "While we can convert most of the Size values to numeric in the same way we converted the Reviews values, there is one value that is impossible to convert." a more simple way of doing the stuff:
```Python
#Your code 
print(len(google_play))
Size=len(google_play[google_play['Size']=='Varies with device'])
ratio=Size/len(google_play)
print(ratio)
#Another way 
odd_value = "Varies with device"
google_play[google_play.Size == odd_value].shape[0] / google_play.shape[0]
```
- In this one "First we remove the dollar sign. Do this in the next cell by applying the str.replace function to the column to replace $ with an empty string ('')." , as it is a dataframe you don't need any built def function or lambda, you can apply the method replace to a entire column:
```Python
#Your code 
google_missing_removed['Price']=google_missing_removed['Price'].apply(lambda x: x.replace('$',''))
# Applying directly the replace method 
google_missing_removed['Price Numerical'] = google_missing_removed['Price'].str.replace('$', '')
# If you get a caveat in execution, try to transform the above using loc method.
```
- Try to complete the challenge 2 when you have some free time.